lib/MetaCPAN/Document/Release.pm

```diff
@@ -1148,9 +1148,10 @@ sub modules {
     );
     return unless $ret->{hits}{total};
 
-    my @files = map { single_valued_arrayref_to_scalar($_) }
-        map +{ %{ $_->{fields} }, %{ $_->{_source} } },
-        @{ $ret->{hits}{hits} };
+    my @files = map {
+        single_valued_arrayref_to_scalar( $_->{fields} );
+        +{ %{ $_->{fields} }, %{ $_->{_source} } }
+    } @{ $ret->{hits}{hits} };
```

Comment:
```
    %{ (single_valued_arrayref_to_scalar( $_->{fields} ))[0] },
```
should work.

Alternatively, single_valued_arrayref_to_scalar should be changed from
```
    return $is_arrayref ? $array : @{$array};
```
to
```
    return $is_arrayref ? $array : $array->[0];
```